packages/flutter_tools/lib/src/ios/ios_emulators.dart

```diff
@@ -39,22 +50,29 @@ class IOSEmulator extends Emulator {
 
   @override
   Future<void> launch() async {
-    Future<bool> launchSimulator(List<String> additionalArgs) async {
-      final List<String> args = <String>[
-        'open',
-        ...additionalArgs,
-        '-a',
-        globals.xcode.getSimulatorPath(),
-      ];
-
-      final RunResult launchResult = await globals.processUtils.run(args);
-      if (launchResult.exitCode != 0) {
-        globals.printError('$launchResult');
-        return false;
-      }
-      return true;
-    }
+    await launchSimulator(<String>[]);
+    return _simulator.boot();
+  }
+}
 
+/// Used by IDEs to "Open iOS Simulator" without a specific simulator.
```

Comment:
@DanTup Is this true?  I see https://github.com/flutter/flutter/commit/c0b2e78391530021e920efabc14826defc0fc62a but I don't know where the code that would call whatever the daemon equivalent of `flutter emulator --launch apple_ios_simulator` would live.

Reply:
The daemon equiv is sending an "emulator.launch" command to the device deamon:

https://github.com/flutter/flutter/blob/f37969638dc9d61c0f26ce16cefe9f92104884c9/packages/flutter_tools/lib/src/commands/daemon.dart#L1101

In VS Code we fetch a list of emulators (also via the daemon) and let the user select one. Then we [send that request](https://github.com/Dart-Code/Dart-Code/blob/c4f37e7503226482e118bd6ff03f3dcf488a8a83/src/extension/flutter/flutter_daemon.ts#L172) to the daemon. The iOS simulators are not treated specially - we just assume whatever is returned from the emulator list can have its ID sent back to `emulator.launch`. So in theory, if you're adding multiple emulators to the `flutter emulator` list for the difference iOS simulators, I'd expect them to just show up in VS Code and also be launchable.

(I don't know if Android Studio/IntelliJ uses these paths yet, as they previously had their own emulator launching functionality before `flutter emulators` existed).

Reply:
I don't see any special use of `apple_ios_simulator`.  If you agree, I'll get rid of the placeholder and see if we get any breakage reports.

Reply:
I'm not sure if I totally understand the question.

IIUC, `apple_ios_simulator` is just an arbitrary ID that was assigned to the Emulator returned by `flutter emulators` that represented the iOS simulator. If you try to launch an emulator with that ID, it would locate that instance and call `launch()` on it.

So I don't think there's anything special about that string - if it was changed (that is, changing the `emulatorId` on the `IOSSimulator` class), nothing should break except that the ID returned by `flutter emulators` and then needed to be passed with `--launch` would be different.

What is important, is that there is some iOS emulator returned by `flutter emulators` and that any IDs returned there can be passed back into `--launch`. That's important for IDEs (or at least VS Code), as we use the output of `flutter emulators` (or rather the daemon equiv) to show the user which emulators are available, and then send it back to `--launch` (or again, the daemon equiv) to start it.

Reply:
My concern was that `apple_ios_simulator` was a magic ID that was used in tooling somewhere I wasn't aware of.  On top of this change `flutter emulator --launch apple_ios_simulator` will fail because there is no `apple_ios_simulator` placeholder anymore, so if it was hardcoded somewhere that would be bad news.  And you seem to have made up the original usage of `apple_ios_simulator` in https://github.com/flutter/flutter/commit/c0b2e78391530021e920efabc14826defc0fc62a so I was asking if you are aware of any cases like that.

Sounds like that wasn't the intention of the magic ID `apple_ios_simulator` though, so we're good.

Reply:
Ah, yes - the ID was just arbitrary (although having `ios` and `apple` in it meant you could do `flutter emulator --launch ios` due to partial matching?) and was not expected to be hard-coded anywhere.

Is the emulator list much larger after this? (I can't remember how many different devices there were and don't have Xcode installed on this machine). I can test everything still looks/works right in VS code after this lands.

Reply:
It's definitely larger after this change, I updated the description with the non-machine output. 
> I can test everything still looks/works right in VS code after this lands.

That would be awesome, thank you!

Reply:
Merged.  Let me know if you find anything, and thanks again!

Reply:
Looks like the `platformType` in the daemon response from `emulator.getEmulators` ([docs](https://github.com/flutter/flutter/blob/master/packages/flutter_tools/doc/daemon.md#emulatorgetemulators)) isn't right. It should be "ios" but is coming through as "iOS-14-3".

We compare that string to the output of `daemon.getSupportedPlatforms` to know which devices/emulators are valid for the current project, so this means none of them are showing up right now. Here's a sample of the JSON:

```
==> [
	{
		"id": "5",
		"method": "emulator.getEmulators"
	}
]
<== [
	{
		"id": "5",
		"result": [
			{
				"id": "C8BCDEC5-8397-433E-A9D6-68FD4B11A180",
				"name": "iPhone 8",
				"category": "mobile",
				"platformType": "iOS-14-3" // bad
			},
			{
				"id": "70EFA6C8-4A4E-4684-9D5C-77879CB95806",
				"name": "iPhone 8 Plus",
				"category": "mobile",
				"platformType": "iOS-14-3" // bad
			},
			// ...
			{
				"id": "flutter_emulator_2",
				"name": "flutter emulator 2",
				"category": "mobile",
				"platformType": "android" // good
			}
		]
	}
]
```